SmartDeviceLink/SDLLockScreenManager.m

```diff
@@ -147,6 +160,46 @@ - (void)sdl_checkLockScreen {
     }
 }
 
+- (void)sdl_toggleLockscreenDismissalableState {
+    BOOL lastLockScreenDismissableEnabled = [self.lastDriverDistractionNotification.lockScreenDismissalEnabled boolValue];
+    if (self.lastDriverDistractionNotification == nil || self.lastDriverDistractionNotification.lockScreenDismissalEnabled == nil ||
+        ![self.lastDriverDistractionNotification.lockScreenDismissalEnabled boolValue]) {
+        self.lockScreenDismissableEnabled = NO;
+    } else {
+        self.lockScreenDismissableEnabled = YES;
+    }
+    
+    if (lastLockScreenDismissableEnabled != self.lockScreenDismissableEnabled) {
+        [self sdl_updateLockscreenDismissalableWithState:self.lockScreenDismissableEnabled];
+    }
+}
+
+- (void)sdl_updateLockscreenDismissalableWithState:(BOOL)enabled {
+    if (![self.lockScreenViewController isKindOfClass:[SDLLockScreenViewController class]]) {
+        return;
+    }
+    
+    __weak typeof(self) weakself = self;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        __strong typeof(self)strongSelf = weakself;
+        if (enabled) {
+            [(SDLLockScreenViewController *)strongSelf.lockScreenViewController addSwipeGestureWithCallback:^{
```

Comment:
You should probably just get a reference to the `lockScreenViewController` and use that instead of continually casting.